plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -3756,6 +3760,42 @@ public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
         commitUpdateAndTransaction(appendFiles, session, transaction, "refresh materialized view");
         transaction = null;
         fromSnapshotForRefresh = Optional.empty();
+
+        // cleanup old snapshot
+        if (snapshotIdBeforeRefreshCommit != null) {
+            TrinoFileSystem fileSystem = fileSystemFactory.create(session.getIdentity(), icebergTable.io().properties());
```

Comment:
There is code duplication here with `io.trino.plugin.iceberg.IcebergMetadata#executeExpireSnapshots`. We should extract out the common part and reuse it here.